src/main/java/software/amazon/awssdk/crt/checksums/Crc.java

```diff
@@ -0,0 +1,23 @@
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+package software.amazon.awssdk.crt.checksums;
+
+import software.amazon.awssdk.crt.CrtResource;
+
+public abstract class Crc extends CrtResource {
+
+    /* java built in implementation of crc32 and crc32c returns a long, this can cause some confusion if not handled
+       properly as crc32 and crc32c only result in 32 bits. be careful when mixing the built in implementation with this
+       one */
+    public static int crc32(byte[] input) {
```

Comment:
why not expose `previous`?

Reply:
native function is public, I'm overloading here to make it an optional argument.